File: backend/internal/model/mission.go
package model

import "time"

// MissionStatus represents the state of a user's mission.
type MissionStatus string

const (
	MissionStatusActive    MissionStatus = "active"
	MissionStatusCompleted MissionStatus = "completed"
	MissionStatusExpired   MissionStatus = "expired"
)

// UserMission represents an observation mission assigned to a user.
type UserMission struct {
	ID          string        `json:"id"`
	UserID      string        `json:"user_id"`
	MissionID   string        `json:"mission_id"`
	CardID      string        `json:"card_id"`
	Title       string        `json:"title"`
	Description string        `json:"description"`
	Status      MissionStatus `json:"status"`
	ValidFrom   time.Time     `json:"valid_from"`
	ValidTo     time.Time     `json:"valid_to"`
	CompletedAt *time.Time    `json:"completed_at,omitempty"`
	CreatedAt   time.Time     `json:"created_at"`
}

// CompleteMissionRequest is the request body for completing a mission.
type CompleteMissionRequest struct {
	UserID    string  `json:"user_id"`
	Latitude  float64 `json:"lat"`
	Longitude float64 `json:"lng"`
}

// CompleteMissionResponse is returned after successfully completing a mission.
type CompleteMissionResponse struct {
	Mission   UserMission `json:"mission"`
	BonusCard *Card       `json:"bonus_card,omitempty"`
}

File: frontend/src/types/mission.ts
export type MissionStatus = "active" | "completed" | "expired";

export interface UserMission {
  id: string;
  user_id: string;
  mission_id: string;
  card_id: string;
  title: string;
  description: string;
  status: MissionStatus;
  valid_from: string;
  valid_to: string;
  completed_at?: string;
  created_at: string;
}

export interface CompleteMissionRequest {
  user_id: string;
  lat: number;
  lng: number;
}

// H4: bonus_card matches backend model.Card (all fields from JSON serialization)
export interface CompleteMissionResponse {
  mission: UserMission;
  bonus_card?: {
    id: string;
    name: string;
    category: string;
    reading_text: string;
    image_url: string;
    description: string;
    magnitude?: number;
    distance?: string;
    best_season: string;
    created_at: string;
  };
}
